Using hydra for multiple nodes

theworkerant opened the pull request:
The application I've inherited does lots of Facebook pulls for many nodes at a time. The performance, as you can imagine, sucks. I took some graph nodes I typically grab and checked out the performance with Benchmark. 

Let me know what you think.

Using ids.each { |id| api.node(id) }

```
Nodes count: 101
  0.130000   0.200000   0.330000 ( 12.652375)
```

Using nodes(ids)

```
Nodes count: 101
  0.130000   0.090000   0.220000 (  3.444642)
```

Commit:
added #nodes using Hydra for pulling lots of graph nodes

theworkerant commented on lib/doesopengraph/graph_api.rb:
```diff
@@ -11,21 +11,44 @@ module DoesOpenGraph
+      end
+      
+      hydra.run
+      responses
```
These probably won't be ordered correctly, but whatever.

theworkerant replied:
fixed on 3002adb, returns hash instead

mccolin replied:
This looks pretty good. I've been meaning to add proper support for ids but haven't gotten around to it, because my applications haven't needed that support, yet.

Let me take a further look and integrate your change when I'm next updating, sometime soon.

Commit:
#nodes returns Hash with keys of input ids

Commit:
took out path from #nodes in error response

Commit:
sometimes Graph returns "false" for some ids, put in proper error for this.

theworkerant committed:
Rolling back all to attack weird Facebook problem

Merge remote-tracking branch 'awexome/master'

Conflicts:
	lib/doesopengraph/graph_api.rb

theworkerant committed:
added in more error info

theworkerant committed:
updated gemspec

theworkerant committed:
reverted gemspec
